### PR TITLE
Add Design support library module

### PR DESCRIPTION
**Pull request by MichaelEvans**
Should address #166 

Is there any guidelines for knowing if I've covered all the APIs / methods that I need to for a given class? I just sorta guessed at ones that people might actually use (IE: Does there need to be a `getBackgroundTintList()` for FAB?)

### CODE REVIEW

**Comment by MichaelEvans** [comment]
Probably need to update the README too, but I guess I should wait to see if there's feedback first.

File: settings.gradle [hunk 1]
```diff
@@ -9,3 +9,4 @@ include ':assertj-android-palette-v7'
 include ':assertj-android-play-services'
 include ':assertj-android-recyclerview-v7'
 include ':assertj-android-support-v4'
+include ':assertj-android-design'
```
Comment (JakeWharton) [inline settings.gradle]: Alphabetize

File: build.gradle [hunk 2]
```diff
@@ -30,6 +30,7 @@ ext {
   supportMediaRouter = 'com.android.support:mediarouter-v7:22.2.1'
   supportPalette = 'com.android.support:palette-v7:22.2.1'
   supportRecyclerView = 'com.android.support:recyclerview-v7:22.2.1'
+  supportDesign = 'com.android.support:design:22.2.1'
```
Comment (JakeWharton) [inline build.gradle]: Alphabetize

**Comment by JakeWharton** [comment]
There are no guidelines. I also just used best judgement when creating the assertions. Address the nits above and we'll get this merged.

File: assertj-android-design/src/main/java/org/assertj/android/design/api/widget/NavigationViewAssert.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package org.assertj.android.design.api.widget;
+
+import android.graphics.drawable.Drawable;
+import android.support.design.widget.NavigationView;
+import org.assertj.core.api.AbstractAssert;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Assertions for {@link NavigationView} instances. */
+public class NavigationViewAssert extends AbstractAssert<NavigationViewAssert, NavigationView> {
```
Comment (JakeWharton) [inline assertj-android-design/src/main/java/org/assertj/android/design/api/widget/NavigationViewAssert.java]: This should extend from `AbstractFrameLayoutAssert`

File: assertj-android-design/src/main/java/org/assertj/android/design/api/widget/TabLayoutAssert.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
+package org.assertj.android.design.api.widget;
+
+import android.support.design.widget.TabLayout;
+import org.assertj.core.api.AbstractAssert;
+
+import static org.assertj.android.internal.IntegerUtils.buildNamedValueString;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Assertions for {@link TabLayout} instances. */
+public class TabLayoutAssert extends AbstractAssert<TabLayoutAssert, TabLayout> {
```
Comment (JakeWharton) [inline assertj-android-design/src/main/java/org/assertj/android/design/api/widget/TabLayoutAssert.java]: This should extend from `HorizontalScrollViewAssert`

**Comment by MichaelEvans** [comment]
I can squash this if it looks good, left it as separate commits in order to make changes more obvious.

File: assertj-android/src/main/java/org/assertj/android/api/widget/AbstractHorizontalScrollViewAssert.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class AbstractHorizontalScrollViewAssert<S extends AbstractHorizontalScrollViewAssert<S, A>, A extends HorizontalScrollView>
+    extends AbstractFrameLayoutAssert<S, A> {
+  protected AbstractHorizontalScrollViewAssert(A actual, Class<S> selfType) {
+    super(actual, selfType);
+  }
+
+  public AbstractHorizontalScrollViewAssert hasMaximumScrollAmount(int amount) {
```
Comment (JakeWharton) [inline assertj-android/src/main/java/org/assertj/android/api/widget/AbstractHorizontalScrollViewAssert.java]: These should return `S` and need to use the `self` trick (copy/paste from other abstract ones).

**Comment by JakeWharton** [comment]
One last thing. Looks good otherwise.

**Comment by MichaelEvans** [comment]
Updated and squashed.

**Comment by JakeWharton** [comment]
Thanks!

**Comment by MichaelEvans** [comment]
Thanks! You can also close #166 now

**Pull request merged**